### PR TITLE
Replace boost by std

### PR DESCRIPTION
**Pull request by sloriot**
## Summary of Changes

Replace features in boost by features introduces in c++11/c++14

### CODE REVIEW

**Comment by sloriot** [comment]
> It seems you need to check the included headers.

I was using `std::empty` and it is available only in c++17

File: Polyline_simplification_2/include/CGAL/Polyline_simplification_2/simplify.h [hunk 1]
```diff
@@ -154,11 +155,11 @@ class Polyline_simplification_2
       (*it)->set_removable(false);
       ++it;
       for(; it != ite; ++it){
-        if((std::next(it) != ite) && (std::prev(it)== std::next(it))){
+        if((boost::next(it) != ite) && (boost::prior(it)==boost::prior(it))){
```
Comment (lrineau) [inline Polyline_simplification_2/include/CGAL/Polyline_simplification_2/simplify.h:158]: Any idea why the behavior is different?

Comment (sloriot) [reply Polyline_simplification_2/include/CGAL/Polyline_simplification_2/simplify.h:158]: The iterator implementation is most probably buggy (some platforms complains that it is not bidirectional and there are infinite loops on other platforms).

**Comment by sloriot** [comment]
Successfully tested in `CGAL-6.0-Ic-15`

**Pull request merged**